lib/CMakeLists.txt

```diff
@@ -44,124 +38,173 @@ add_library(arango_static_strings STATIC
+        Basics/ArangoGlobalContext.cpp
+        Basics/CleanupFunctions.cpp
+        Basics/ConditionLocker.cpp
+        Basics/ConditionVariable.cpp
+        Basics/ErrorCode.cpp
+        Basics/Exceptions.cpp Basics/Exceptions.tpp Basics/Exceptions.h
+        Basics/FileResult.cpp
+        Basics/FileUtils.cpp
+        Basics/Guarded.h
+        Basics/Identifier.cpp
+        Basics/Mutex.cpp
+        Basics/NumberOfCores.cpp
+        Basics/PageSize.cpp
+        Basics/PhysicalMemory.cpp
+        Basics/ReadWriteLock.cpp
+        Basics/RebootId.cpp Basics/RebootId.h
+        Basics/Result.cpp
+        Basics/ResultError.cpp Basics/ResultError.tpp Basics/ResultError.h
+        Basics/StringBuffer.cpp
+        Basics/StringHeap.cpp
+        Basics/StringUtils.cpp
+        Basics/Thread.cpp
+        Basics/UnshackledMutex.cpp
+        Basics/application-exit.cpp
+        Basics/conversions.cpp
+        Basics/debugging.cpp
+        Basics/error.cpp
+        Basics/files.cpp
+        Basics/fpconv.cpp
+        Basics/levenshtein.cpp
+        Basics/memory.cpp
+        Basics/process-utils.cpp
+        Basics/signals.cpp
+        Basics/system-functions.cpp
+        Basics/tri-strings.cpp
+        Containers/ImmerMemoryPolicy.h
+        Logger/Escaper.cpp
+        Logger/LogAppender.cpp
+        Logger/LogAppenderFile.cpp
+        Logger/LogAppenderSyslog.cpp
+        Logger/LogContext.cpp
+        Logger/LogLevel.cpp
+        Logger/LogThread.cpp
+        Logger/LogTopic.cpp
+        Logger/Logger.cpp
+        Logger/LoggerFeature.cpp
+        Logger/LoggerStream.cpp
+        Logger/LogTimeFormat.cpp
+        ProgramOptions/IniFileParser.cpp
+        ProgramOptions/Option.cpp
+        ProgramOptions/Parameters.cpp
+        ProgramOptions/ProgramOptions.cpp
+        ProgramOptions/Section.cpp
+        ProgramOptions/Translator.cpp
+        ProgramOptions/UnitsHelper.cpp
+        Random/RandomGenerator.cpp
+)
+
+if (MSVC)
+  target_sources(arango_lightweight PRIVATE
+          Basics/locks-win32.cpp
+          Basics/terminal-utils-win32.cpp
+          Basics/threads-win32.cpp
+          )
+else ()
+  target_sources(arango_lightweight PRIVATE
+          Basics/locks-posix.cpp
+          Basics/terminal-utils-posix.cpp
+          Basics/threads-posix.cpp
+          )
+endif ()
+
+target_include_directories(arango_lightweight PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+# some auto-generated headers live here, like "Basics/build.h"
+target_include_directories(arango_lightweight PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
+target_include_directories(arango_lightweight PUBLIC
+        ${FROZEN_INCLUDE_DIR}
+        ${FUNCTION2_INCLUDE_DIR}
+        )
+# Rebuild generated error files so they are up-to-date when all
+# dependent stuff is built
+add_dependencies(arango_lightweight errorfiles)
+add_dependencies(arango_lightweight exitcodefiles)
+
+target_link_libraries(arango_lightweight PUBLIC velocypack PRIVATE date_interface fmt
```

Comment:
Is some difference between fmt and fmt::fmt?

I see libarango link against second